proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
     -> UnsafeMutablePointer<T>
 
-  func initialize<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func initializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
   // The `move` APIs deinitialize the memory at `from`.
-  func moveInitialize<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func moveInitializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
-  func load<T>(_: T.Type) -> T
-  func load<T>(_: T.Type, atByteOffset: Int) -> T
-  func load<T>(fromContiguous: T.Type, atIndex: Int) -> T
+  func load<T>(as: T.Type) -> T
+  func load<T>(fromByteOffset: Int, as: T.Type) -> T
+  func load<T>(fromContiguous: T.Type, at: Int) -> T
```

Comment:
on Fri Jul 08 2016, Andrew Trick <notifications-AT-github.com> wrote:

>>      -> UnsafeMutablePointer<T>
>>  
>> -  func load<T>(_: T.Type) -> T
>> -  func load<T>(_: T.Type, atByteOffset: Int) -> T
>> -  func load<T>(fromContiguous: T.Type, atIndex: Int) -> T
>> +  func load<T>(as: T.Type) -> T
>> +  func load<T>(fromByteOffset: Int, as: T.Type) -> T
>> +  func load<T>(fromContiguous: T.Type, at: Int) -> T
>
> @jckarter requested the load-from-byte-offset feature as a significant
> improvement over `(p + offset).load(as:)`. I think Joe's main argument
> is that if `offset` is a complex expression, the `fromByteOffset` form
> will be much more readable. I agree.

We have normal ways of dealing with that situation:

    let byteOffset = a complex expression
    (p + byteOffset).load(as: Int.self)

There's no need to design special provisions into this API for doing
what every Swift programmer knows how to do.

> I'm not particularly interested in very seldom used, purely additive
> APIs, but in this case I do think it the `byteOffset` form, for both
> load and store, makes it more clear that offset is in bytes and that
> the address computation is untyped. Otherwise, the user needs to infer
> that all from the pointer type.

Then are we planning to omit arithmetic on UnsafeRawPointer altogether?
If not, why not?